gmso/formats/gro.py

```diff
@@ -131,7 +134,8 @@ def write_gro(top, filename, precision=3):
 
     """
     pos_array = np.ndarray.copy(top.positions)
-    pos_array = _validate_positions(pos_array)
+    if shift_coord:
+        pos_array = _validate_positions(pos_array)
```

Comment:
I was glancing at this function and feel like it should be rewritten to use vectorized operations instead of this a for loop.

```python
def _validate_positions(pos_array):
    """Modify coordinates, as necessary, to fit limitations of the GRO format."""
    if np.min(pos_array) < 0:
        warnings.warn(
            "Topology contains some negative positions. Translating "
            "in order to ensure all coordinates are non-negative."
        )
    min_xyz = np.min(x, axis=1) # smallest position of each coordinate
    min_xyz0 =  np.where(min_xyz<0, min_xyz, 0) #shift by 0 or minimum value of the positions in that coordinate
    pos_array -= min_xyz0
    return pos_array
```

Reply:
Same as below comment. I think it's worthwhile to make these changes to this function as well.

Reply:
oh shoot, I somehow miss these, I will add these changes in